src/Fls.Results/OperationResult.cs

```diff
@@ -105,20 +191,40 @@ public static IOperationResult<T> Failure<T>(Exception exception)
             );
         }
 
+        /// <summary>
+        /// Bindes exactly errors and failures. In case of error of failure calls bind, other ways passes previous result forward
+        /// </summary>
+        /// <param name="source"></param>
+        /// <param name="bind">Delegate to bind error of failure</param>
+        /// <param name="getMessage">Delegetate to find out error message for given exception</param>
+        /// <returns>Binned error/failure or result of previous operation if is was succeeded</returns>
         public static IOperationResult<T> BindError<T>(this IOperationResult<T> source, Func<string, IOperationResult<T>> bind, Func<Exception, string> getMessage)
         {
             return source.BindError((_, error) => bind(error), getMessage);
         }
 
+        /// <summary>
+        /// Bindes exactly errors and failures. In case of error of failure calls bind, other ways passes previous result forward. 
```

Comment:
Suggested changing to:

If <code>source</code> matches as a success it is passed on, if not the error message/code pair is bound with <code>bind</code>. If <code>source</code> matches as a failure, the <code>bind</code> function is called with an error message extracted from the attached Exception using <code>getErrorMessage</code> and the one-for-all critical failure code <code>exceptionCode</code>.